### PR TITLE
Feature/arf protocol integration

### PR DESCRIPTION
**Pull request by neeeekitos**

### CODE REVIEW

**Comment by neeeekitos** [comment]
@msaug, arf integrated + some fixes for token amounts when adding a liquidity

File: src/components/addv2/Add.tsx [hunk 1]
```diff
@@ -255,7 +253,10 @@ const ActionBlockAdd = (props: ActionBlockProps) => {
     });
     addToken(token0Selector.name, token0);
     addToken(token1Selector.name, token1);
-    const tokenAmountFrom = new TokenAmount(token0, ethers.utils.parseUnits(amountToken0, token0.decimals).toString());
+    const tokenAmountFrom = new TokenAmount(token0,
+      ethers.utils.parseUnits(Number(amountToken0).toFixed(20), token0.decimals).toString());
```
Comment (msaug) [inline src/components/addv2/Add.tsx:257]: (Number(amountToken0).toFixed(20) => why ?
Is it for scientific notation ?

**Review by msaug — COMMENTED** [review]
wunderschön

File: src/components/addv2/Add.tsx (hunk 1 above)
Comment (neeeekitos) [reply src/components/addv2/Add.tsx:257]: Sometimes it throws an error. When the number is too small, the exponential notation is used, which ethers can't parse. So we have to convert it to the normal notation before.

**Pull request merged**